File: PYTHON/GATE-FOUR/test_day_four_level_five.py
from unittest import TestCase
from day_four_level_five import *

class TestFile(TestCase):

    def test_that_array_function_returns_sorted_array_of_prime_numbers(self):
        numbers = [5, 9, 3, 6, 2]
        expected = [5, 3, 2]

        actual = get_sorted_prime_numbers(numbers)

        self.assertEqual(expected, actual)

    def test_that_array_function_replaces_all_negative_numbers_with_zero(self):
        numbers = [5, -9, 3, -6, 2, -11]
        expected = [5, 0, 3, 0, 2, 0]

        actual = replace_negative_elements_with_zero(numbers)

        self.assertEqual(expected, actual)

    def test_that_array_function_moves_all_zero_elements_to_end(self):
        numbers = [5, 0, 3, 0, 2, 0]
        expected = [5, 3, 2, 0, 0, 0]

        actual = move_zero_elements_to_end_of_array(numbers)

        self.assertEqual(expected, actual)

    def test_that_array_function_returns_common_elements(self):
        numbers = [45, 60, 3, 0, 67, 2, 45, 3, 22, 0]
        expected = [45, 3, 0]

        actual = get_common_elements(numbers)

        self.assertEqual(expected, actual)

File: JAVA/GATE-FOUR/TestDayFourLevelFive.java

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;

public class TestDayFourLevelFive{

    DayFourLevelFive array;
    
    @BeforeEach
    void setUp(){
    
        array = new DayFourLevelFive();
    }
    
    @Test
    public void testThatArrayFunctionReturnsSortedArrayOfPrimeNumbers(){
    
        int[]numbers = {5,9,3,6,2};
        int[]expected = array.getSortedPrimeNumbers(numbers);
        int[]actual = {5,3,2};
        
        assertArrayEquals(expected,actual);
        
    }
    
    @Test
    public void testThatArrayFunctionReplacesAllNegativeNumbersInAGivenArrayWithZero(){
    
        int[]numbers = {5,-9,3,-6,2,-11};
        int[]expected = array.replaceNegativeElementsWithZero(numbers);
        int[]actual = {5,0,3,0,2,0};
        
        assertArrayEquals(expected,actual);
        
    }
    
    @Test
    public void testThatArrayFunctionMovesAllZeroElementToEndOfArray(){
    
        int[]numbers = {5,0,3,0,2,0};
        int[]expected = array.moveZeroElementsToEndOfArray(numbers);
        int[]actual = {5,3,2,0,0,0};
        
        assertArrayEquals(expected,actual);
        
    }
    
    @Test
    public void testThatArrayFunctionReturnsCommonElementInAnArray(){
    
        int[]numbers = {45,60,3,0,67,2,45,3,22,0};
        int[]expected = array.getCommonElements(numbers);
        int[]actual = {45,3,0};
        
        assertArrayEquals(expected,actual);
        
    }
}
